pypeit/sensfunc.py

```diff
@@ -462,126 +713,273 @@ def write_QA(self):
                                                     polyfunc=self.par['UVIS']['polyfunc'],
                                                     debug=self.debug)
-        # Add the algorithm to the meta_table
-        meta_table['ALGORITHM'] = self.par['algorithm']
-
-        self.steps.append(inspect.stack()[0][3])
-
-        return meta_table, out_table
 
+        # Copy the relevant metadata
+        self.std_name = meta_table['STD_NAME'][0]
+        self.std_cal = meta_table['CAL_FILE'][0]
+        self.std_ra = meta_table['STD_RA'][0]
+        self.std_dec = meta_table['STD_DEC'][0]
+        self.airmass = meta_table['AIRMASS'][0]
+        self.exptime = meta_table['EXPTIME'][0]
+
+        norder, nspec = out_table['SENS_ZEROPOINT'].shape
+
+        # Instantiate the main output data table
+        self.sens = self.empty_sensfunc_table(norder, nspec)
+
+        # Copy the relevant data
+        # NOTE: SENS_COEFF is empty!
+        self.sens['SENS_WAVE'] = out_table['SENS_WAVE']
+        self.sens['SENS_COUNTS_PER_ANG'] = out_table['SENS_COUNTS_PER_ANG']
+        self.sens['SENS_ZEROPOINT'] = out_table['SENS_ZEROPOINT']
+        self.sens['SENS_ZEROPOINT_GPM'] = out_table['SENS_ZEROPOINT_GPM']
+        self.sens['SENS_ZEROPOINT_FIT'] = out_table['SENS_ZEROPOINT_FIT']
+        self.sens['SENS_ZEROPOINT_FIT_GPM'] = out_table['SENS_ZEROPOINT_FIT_GPM']
+        if self.meta_spec['ECH_ORDERS'] is not None:
+            self.sens['ECH_ORDERS'] = self.meta_spec['ECH_ORDERS']
+        self.sens['POLYORDER_VEC'] = np.full(norder, self.par['polyorder'])
+        self.sens['WAVE_MIN'] = out_table['WAVE_MIN']
+        self.sens['WAVE_MAX'] = out_table['WAVE_MAX']
 
     def eval_zeropoint(self, wave, iorddet):
         """
+        Evaluate the sensitivity function zero-points
 
         Parameters
         ----------
-        wave: ndarray shape (nspec)
-           Wavelength array
-
-        iorddet: int
-           Order or detector
+        wave : `numpy.ndarray`_, shape is (nspec)
+            Wavelength array
+        iorddet : :obj:`int`
+            Order or detector (0-indexed)
 
         Returns
         -------
-        zeropoint: ndarray, shape (nspec,)
-
+        zeropoint : `numpy.ndarray`_, shape is (nspec,)
         """
         # This routine can extrapolate
-        zeropoint = scipy.interpolate.interp1d(self.out_table['SENS_WAVE'][iorddet,:], self.out_table['SENS_ZEROPOINT_FIT'][iorddet,:],
-                                              bounds_error = False, fill_value='extrapolate')(wave)
-        return zeropoint
+        return scipy.interpolate.interp1d(self.sens['SENS_WAVE'][iorddet,:],
+                                          self.sens['SENS_ZEROPOINT_FIT'][iorddet,:],
+                                          bounds_error=False, fill_value='extrapolate')(wave)
+
+
+# This was moved from pypeit/core/coadd.py, and appropriate adjustments were
```

Comment:
Should this be a classmethod on SensFunc?

Reply:
Done.